src/deployPanel.ts

```diff
@@ -75,61 +77,122 @@ export class DeployPanel {
+                    .emitPush(this.viewState.contractMetadata.entrypointReturnTypeHex)
+                    .emitPush(this.viewState.contractMetadata.entrypointParameterTypesHex)
+                    .emitPush(this.viewState.contractAvmHex)
+                    .emitSysCall('Neo.Contract.Create')
+                    .str;
+
+                // Determine required GAS:
+                const rpcClient = new neon.rpc.RPCClient(this.rpcUri);
+                const invokeResult = await rpcClient.invokeScript(script);
+                const gas = parseFloat(invokeResult.gas_consumed);
+
+                const walletConfig = this.viewState.wallets.filter(_ => _.address === this.viewState.walletAddress)[0];
+                if (await walletConfig.unlock()) {
+                    const api = new neon.api.neoCli.instance(this.rpcUri);
+                    const config = {
+                        api: api,
+                        script: script,
+                        account: walletConfig.account,
+                        signingFunction: walletConfig.signingFunction,
+                        gas: gas,
+                    };
+                    const result = await neon.default.doInvoke(config);
+                    if (result.response && result.response.txid) {
+                        this.viewState.result = result.response.txid;
+                        this.viewState.showError = false;
+                        this.viewState.showSuccess = true;
+                    } else {
+                        this.viewState.result = 'No response from RPC server; contract may not have deployed';
+                        this.viewState.showError = true;
+                        this.viewState.showSuccess = false;
+                    }
+                }
+            } catch (e) {
+                this.viewState.result = 'Error deploying contract: ' + e;
+                this.viewState.showError = true;
+                this.viewState.showSuccess = false;
+            }
+        }
+    }
+
+    private async ensureMetadata() {
+        const stringFields = [ 'description', 'email', 'author', 'version', 'title' ];
+        const flagFields = [ 
+            [ 'has-storage', 'storage' ],
+            [ 'has-dynamic-invoke', 'dynamic invoke' ],
+        ];
+        const missingStringFields = stringFields.filter(_ => this.viewState.contractMetadata[_] === undefined);
+        const missingFlagFields = flagFields.filter(_ => this.viewState.contractMetadata[_[0]] === undefined);
+        const missingReturnType = !this.viewState.contractMetadata.entrypointReturnTypeHex;
+        const missingParmeterTypes = this.viewState.contractMetadata.entrypointParameterTypesHex === undefined;
+        if (missingStringFields.length || missingFlagFields.length || missingReturnType || missingParmeterTypes) {
+            if (await this.promptForBooleanOrCancel('Some contract metadata is missing. Would you like to provide the missing metadata manually?', false)) {
```

Comment:
we should only prompt for has storage and has dynamic invoke if it's missing from the ABI file. other contract metadata should simply use defaults. Missing return and parameter types indicates a missing or invalid abi.json file, so that should be an error instead of a prompt

Reply:
> we should only prompt for has storage and has dynamic invoke if it's missing from the ABI file

That's the current behavior.

> other contract metadata should simply use defaults

I don't think we can distinguish between someone wanting them to be empty vs. using the old compiler and not having had an opportunity to set them?

> Missing return and parameter types indicates a missing or invalid abi.json file, so that should be an error instead of a prompt

It can also be caused by a missing ABI file. Do we want to disallow deployment of contracts that the user does not have an ABI file for?

Reply:
> > other contract metadata should simply use defaults
> 
> I don't think we can distinguish between someone wanting them to be empty vs. using the old compiler and not having had an opportunity to set them?

Oh, actually we can... How about I assume that:
- if the metadata node is present but contains empty strings, then the user explicitly decided not to publish values
- if the metadata node is missing, the user may want to provide values

Reply:
as far as I can tell, these aren't widely used in the neo dev community. they are being removed in neo 3. I say we use the avm file name for title (w/o extension) and default everything else to empty string if the metadata node is missing. I'd rather err on the side of simple for the developer. Plus, if the developer REALLY wants these, they can always upgrade to the latest tools.